shell/platform/darwin/ios/framework/Source/FlutterUndoManagerPluginTest.mm

```diff
@@ -164,4 +165,38 @@ - (void)testSetUndoStateDoesInteractWithInputDelegate {
   OCMVerify(never(), [self.activeTextInputView inputDelegate]);
 }
 
+- (void)testDeallocRemovesAllUndoManagerActions {
+  __weak FlutterUndoManagerPlugin* weakUndoManagerPlugin;
+  // Use a real undo manager.
+  NSUndoManager* undoManager = [[NSUndoManager alloc] init];
+  @autoreleasepool {
+    id activeTextInputView = OCMClassMock([TextInputViewTest class]);
+
+    FakeFlutterUndoManagerDelegate* undoManagerDelegate =
+        [[FakeFlutterUndoManagerDelegate alloc] initWithUndoManager:undoManager
+                                                activeTextInputView:activeTextInputView];
+
+    FlutterUndoManagerPlugin* undoManagerPlugin =
+        [[FlutterUndoManagerPlugin alloc] initWithDelegate:undoManagerDelegate];
+    weakUndoManagerPlugin = undoManagerPlugin;
+
+    FlutterMethodCall* setUndoStateCall =
+        [FlutterMethodCall methodCallWithMethodName:@"UndoManager.setUndoState"
+                                          arguments:@{@"canUndo" : @YES, @"canRedo" : @YES}];
+    [undoManagerPlugin handleMethodCall:setUndoStateCall
+                                 result:^(id _Nullable result){
+                                 }];
+    XCTAssertTrue(undoManager.canUndo);
+    XCTAssertTrue(undoManager.canRedo);
+    // Fake out the undoManager being nil, which happens when the FlutterViewController deallocs and
+    // the undo manager can't be fetched from the FlutterEngine delegate.
+    undoManagerDelegate.undoManager = nil;
+  }
+  XCTAssertNil(weakUndoManagerPlugin);
+  // Regression test for https://github.com/flutter/flutter/issues/150408.
+  // Undo manager undo and redo stack should be empty after the plugin deallocs.
+  XCTAssertFalse(undoManager.canUndo);
```

Comment:
Wrote a new test that fails on main without this change (though I couldn't get it to actually crash):
```
FlutterUndoManagerPluginTest.mm:198: error: -[FlutterUndoManagerPluginTest testDeallocRemovesAllUndoManagerActions] : ((undoManager.canUndo) is false) failed
```